tests/test_preprocess.py

```diff
@@ -168,11 +168,35 @@ def test_split_raster_empty(config):
     assert os.path.exists("tests/output/empty/OSBS_029_1.png")
 
 
-def test_split_size_error(config):
+def test_split_size_error(config, tmpdir):
     with pytest.raises(ValueError):
         annotations_file = preprocess.split_raster(path_to_raster=config["path_to_raster"],
                                                    annotations_file=config["annotations_file"],
-                                                   base_dir="tests/data/",
+                                                   base_dir=tmpdir,
                                                    patch_size=2000,
                                                    patch_overlap=config["patch_overlap"])
-    
\ No newline at end of file
+
+def test_split_raster_4_band_warns(config, tmpdir):
+    # Confirm that the rasterio channel order is C x H x W
+    assert rasterio.open(get_data("OSBS_029.tif")).read().shape[0] == 3
+
+    # Create a 4 band image in the wrong channel order, it should be H x W x C
+    numpy_image = np.zeros((4, 400, 400), dtype=np.uint8)
+    
+    with pytest.warns(UserWarning):
```

Comment:
@ Ben Why are we repeating the procedure ?

Reply:
Once for when the channel order channels x height x width and once for when the channel order is height x width x channels. The function should work and warn users in either situation. Before it was assuming the correct shape, which is channel x height x width, without actually checking it was true.